File: Extra - Estudos/2174 - ColecaoPokemon/Main.java
import java.util.*;
public class Main {

    public static int sortPokemons(String[] p, int cap){

        int iguais = 0;
        for(int i = 0; i < cap - 1; i++){
            int menor = i;
            for(int j = i + 1 ; j < cap ; j++){
                if(p[j].compareTo(p[menor]) < 0){
                    menor = j;
                } else if(p[i].compareTo(p[i + 1]) == 0){
                    iguais++;
                }
            }
            if(menor != i){
                String tmp = p[i]; 
                p[i] = p[menor];
                p[menor] = tmp;
            }
        }
        return iguais;

    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int capturados = sc.nextInt();
        sc.nextLine();

        String[] pokemon = new String[capturados];

        for(int i = 0 ; i < capturados ; i++){
            String nomePokemon = sc.nextLine();
            pokemon[i] = nomePokemon;
        }

        int same = sortPokemons(pokemon, capturados);

       int pokemonNaoRepetidos = capturados - same;
       int resp = 151 - pokemonNaoRepetidos;

       System.out.println("Falta(m) " + resp + " pomekon(s).");


        sc.close();
    }
    
}

File: Extra - Estudos/2174 - ColecaoPokemon/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int sortPokemons(char p[][21], int cap) {
    int iguais = 0;

    for (int i = 0; i < cap - 1; i++) {
        int menor = i;
        for (int j = i + 1; j < cap; j++) {
            if (strcmp(p[menor], p[j]) > 0) {
                menor = j;
            }
        }
        if (menor != i) {
            char tmp[21];
            strcpy(tmp, p[i]);
            strcpy(p[i], p[menor]);
            strcpy(p[menor], tmp);
        }
    }

    for (int i = 0; i < cap - 1; i++) {
        if (strcmp(p[i], p[i + 1]) == 0) {
            iguais++;
        }
    }

    return iguais;
}

int main() {
    int capturados;
    scanf("%d", &capturados);

    char nomesPokemons[capturados][21];

    for (int i = 0; i < capturados; i++) {
        scanf("%s", nomesPokemons[i]);
    }

    int same = sortPokemons(nomesPokemons, capturados);
    int semRepetidos = capturados - same; 
    int answer = 151 - semRepetidos; 

    printf("Falta(m) %d pomekon(s).\n", answer);

    return 0;
}
